vector/src/main/java/im/vector/riotx/features/home/ShortcutsHandler.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.riotx.features.home
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.os.Build
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import im.vector.matrix.android.api.session.room.model.tag.RoomTag
+import im.vector.matrix.android.api.util.toMatrixItem
+import im.vector.riotx.core.glide.GlideApp
+import im.vector.riotx.core.utils.DimensionConverter
+import im.vector.riotx.features.home.room.detail.RoomDetailActivity
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+private val useAdaptiveIcon = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
+private const val adaptiveIconSizeDp = 108
+private const val adaptiveIconOuterSidesDp = 18
+
+class ShortcutsHandler @Inject constructor(
+        private val homeRoomListStore: HomeRoomListDataSource,
+        private val avatarRenderer: AvatarRenderer,
+        private val dimensionConverter: DimensionConverter
+) {
+
+    @SuppressLint("CheckResult")
+    fun observeRoomsAndBuildShortcuts(context: Context) {
```

Comment:
Remove context parameter and return the disposable. The stream has to be disposed to avoid memory leak. It will be disposed once the Activity is destroyed (see other comment)

Reply:
Ah, great idea 👌 this is much better